bin/publish-npm-modules.sh

```diff
@@ -63,15 +78,27 @@ mkdir -p "$SOCKET_HOME"
 export SOCKET_HOME
 export PREFIX
 
-if (( do_global_link && !dry_run )); then
-  echo >&2 "warning: '--link' implies '--dry-run' too"
-  dry_run=1
+# Confirm that ssc in PATH matches current commit
+if command -v ssc >/dev/null 2>&1; then
+  REPO_VERSION="$(cat "$root/VERSION.txt") ($(git rev-parse --short=8 HEAD))"
+  PATH_VERSION=$(ssc --version)
+
+  if [[ "$REPO_VERSION" != "$PATH_VERSION" ]]; then
+    echo "Repo $REPO_VERSION and $(which ssc) $PATH_VERSION don't match."
+    exit 1
+  fi
 fi
 
-if (( !only_top_level )); then
+if (( !only_top_level ))  && (( !no_rebuild )) ; then
   "$root/bin/install.sh" || exit $?
 fi
 
+if [[ "$(find $SOCKET_HOME -name lib*.a | grep android | wc -l)" -lt "8" ]]; then
```

Comment:
This seems to brittle - why are we checking lib count? This is also special treatment for android

Reply:
I implemented the check for android because:
1. There's more effort involved in setting it up
2. ios libs are pretty much out of the box, so there's a low probability they won't build
3. You can't do much without `ssc` building, so unlikely that someone would try to ship a package without desktop libs and `ssc` bin.
4. Android adds considerably more workload to the build. I envisage switching off android builds (and IOS eventually) for optimised build runs catered to the current build target.

If I had implemented an option for IOS libs being excluded from build, I would have also added this check.

You raise a good point, I'm all for implementing these checks across the board.

Reply:
why can't we verify that the files we care about exist instead of counting how many there are?

Reply:
Resolved as discussed.